src/storage/AsyncLockResourceLocker.ts

```diff
@@ -0,0 +1,29 @@
+import AsyncLock from 'async-lock';
+import { Lock } from './Lock';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { ResourceLocker } from './ResourceLocker';
+
+/**
+ * A resource locker making use of the `async-lock` library.
+ */
+export class AsyncLockResourceLocker implements ResourceLocker {
```

Comment:
Is this implementation multi-worker-safe, or even thread-safe?
We might want to expose that information as two boolean properties.

Reply:
Multiple threads in Node means multiple independent processes right? Which would mean multiple instances of this object so they would be completely independent (and thus the locking would not work between the workers). Unless workers threads work differently than I though.

Reply:
It depends on the locking mechanism used. Workers are a multi-process mechanism indeed.
I think we probably want a `threadSafe` flag or so (but we will likely not need the two, as I said earlier).

Suggestion to perhaps rename the class to `SingleThreadedResourceLocker` or something (the lib used for implementation doesn't really matter).